tests/periph/fmc/app.config.test

```diff
@@ -0,0 +1,3 @@
+CONFIG_MODULE_BENCHMARK=y
+CONFIG_MODULE_PERIPH_FMC=y
```

Comment:
To handle the FEATURES_OPTIONAL as defined in the Makefile, you have to add a Kconfig file in the application directory with the following content:

```kconfig
config APPLICATION
    bool
    default y
    imply MODULE_PERIPH_FMC_NOR_SRAM
    imply MODULE_PERIPH_FMC_SDRAM
    depends on TEST_KCONFIG
```

I checked in tests/periph/rtc and tests/periph/rtt where a similar pattern is used.

Reply:
It also seems to work without this extra Kconfig file because of the following in `cpu/stm32/periph/Kconfig.fmc`:

```python
config MODULE_PERIPH_FMC_NOR_SRAM
    bool "NOR/PSRAM/SRAM support"
    default y if HAS_PERIPH_FMC_NOR_SRAM
    help
        Enables NOR Flash, PSRAM and SRAM support of the STM32 FMC/FSMC
        driver.

config MODULE_PERIPH_FMC_SDRAM
    bool "SDRAM support"
    default y if HAS_PERIPH_FMC_SDRAM
    help
        Enables SDRAM support of the STM32 FMC/FSMC driver.
```

Reply:
I had the feeling that `default y if HAS_*` is the way to model same behavior as `FEATURES_OPTIONAL`. I'm really at war with Kconfig. I'm always happy when it somehow gives the same result as the Makefiles.

Reply:
I insist on doing it like I proposed in https://github.com/RIOT-OS/RIOT/pull/19843/files#r1273159870

Reply:
:thinking: I see. But isn't that what a user expects when he says `CONFIG_PERIPH_FMC=y` that the board enables the features it has by default? He can unselect it using `menuconfig`. Or would the user have to say `CONFIG_PERIPH_FMC_SDRAM=y` for example?

As said, I'm really at war with Kconfig. It makes the life much harder from my point of view :wink:

Reply:
> I'm really at war with Kconfig. It makes the life much harder from my point of view

I have the same feeling sometimes. But I think the main problem is that we are still midstream with the migration to Kconfig and it is moving forward too slowly (almost stuck I'd say), which is a pain.

Reply:
BTW, while crawling through some `tests/*/Kconfig`s to understand our guidelines a bit better, I saw in `tests/pkg/qr-code-generator/Kconfig` that it lists explicitly the boards that have a display. I would have to extend it a lot :see_no_evil: The list is by far not complete. Isn't there a more intelligent solution, for example defining `HAVE_DISPLAY` as a feature the boards select?

Reply:
> Isn't there a more intelligent solution, for example defining `HAVE_DISPLAY` as a feature the boards select?

That doesn't exist yet but it would be nice to have I think

Reply:
> I insist on doing it like I proposed in https://github.com/RIOT-OS/RIOT/pull/19843/files#r1273159870

I realized it in that way.